### PR TITLE
Add unit tests for cdbdisp_dispatchPlan and cdbdisp_makeResult.

### PR DESCRIPTION
**Pull request by eurekaka**

### CODE REVIEW

File: src/backend/cdb/dispatcher/test/cdbdispatchresult_test.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+test__cdbdisp_makeResult__oom(void **state)
+{
+	CdbDispatchResult *result = NULL;
+
+	struct CdbDispatchResults *results = _init_cdbdisp_makeResult();
+	struct SegmentDatabaseDescriptor *segdbDesc =
+		(struct SegmentDatabaseDescriptor *) palloc0(sizeof(struct SegmentDatabaseDescriptor));
+
+	will_return(__wrap_createPQExpBuffer, NULL);
+	will_be_called(__wrap_destroyPQExpBuffer);
```
Comment (xiong-gang) [inline src/backend/cdb/dispatcher/test/cdbdispatchresult_test.c]: it would be better to have some comments here.

File: src/backend/mock.mk [hunk 2]
```diff
@@ -58,9 +58,16 @@ ALL_OBJS=$(addprefix $(top_srcdir)/, \
...
+
+WRAP_SED_REGEXP='s/.*__wrap_\(\w*\)(.*/\1/p'
+
+WRAP_FUNCS=$(addprefix $(WRAP_FLAGS), \
+			$(sort $(shell sed -n $(WRAP_SED_REGEXP) $(1))))
+
 # The test target depends on $(OBJFILES) which would update files including mocks.
 %.t: $(OBJFILES) $(CMOCKERY_OBJS) $(MOCK_OBJS) %_test.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $(call BACKEND_OBJS, $(top_srcdir)/$(subdir)/$*.o $(patsubst $(MOCK_DIR)/%_mock.o,$(top_builddir)/src/%.o, $^)) $(filter-out %/objfiles.txt, $^) $(MOCK_LIBS) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $(call WRAP_FUNCS, $(top_srcdir)/$(subdir)/test/$*_test.c) $(call BACKEND_OBJS, $(top_srcdir)/$(subdir)/$*.o $(patsubst $(MOCK_DIR)/%_mock.o,$(top_builddir)/src/%.o, $^)) $(filter-out %/objfiles.txt, $^) $(MOCK_LIBS) -o $@
```
Comment (xiong-gang) [inline src/backend/mock.mk]: add some comments please.

**Comment by xiong-gang** [comment]
LGTM

**Comment by danielgustafsson** [comment]
LGTM too, but I second @xiong-gang's request for comments. Nice to see improved coverage through proper mock tests, great!

**Comment by eurekaka** [comment]
Merged, thanks for reviewing.

**Pull request closed without merging**